### DISCUSSION TITLE
Autorizzazione necessaria per la creazione di una fattura

### DISCUSSION THREAD

**Discussion by damcompany** [fattureincloud/api#315] [Q&A] [answered]
Buon pomeriggio, sto modificando il collegamento tramite API passando dalla versione 1 alla 2, di un software sviluppato con PHP.
Ho visto che ora bisogna creare un'app ed utilizzare oauth per collegarla al software.
L'unico dubbio riguarda l'autorizzazione che deve essere per forza fatta nel browser da parte dell'utente e non può essere quindi fatta direttamente in background come avveniva prima?

Ovvero l'utente clicca su "Crea fattura su Fattureincloud" , viene generato l'url con gli scopes necessari e l'utente viene reindirizzato sul browser nella schermata di Fattureincloud per accettare la richiesta e successivamente la fattura viene creata?
Oppure c'è un modo per far si che in background, PHP con una chiamata curl recuperi l'access token che è necessario poi per lanciare le chiamate API necessarie?

**Comment by emajo** [comment, marked as answer]
Ciao @damcompany, oauth non è l'unico metodo di autenticazione che mettiamo a disposizione, abbiamo anche l'autenticazione con token manuale, [questo schema](https://developers.fattureincloud.it/docs/basics/authentication#%EF%B8%8F-which-authentication-should-i-use) può aiutare nella scelta del medoto corretto per i vari casi d'uso.
In caso sia effettivamente oauth il metodo corretto per il suo caso d'uso, l'utente dovrà compiere lo step di login con Fatture in Cloud solo la prima volta, in quanto dopo questo passaggio si ottiene un refresh token che può essere usato in background per ottenere un nuovo access token.

Grazie,
Emanuele.

**Reply by valmoz** [reply]
Buongiorno @damcompany,
abbiamo risposto alla sua domanda?

Grazie
Mauro
